crates/turborepo-lib/src/diagnostics.rs

```diff
@@ -0,0 +1,523 @@
+                                    join_all(futures).await.into_iter().collect();
+
+                                match results {
+                                    Ok(_) => {
+                                        Self::log_line(
+                                            &chan,
+                                            "Git FS Monitor settings set".to_string(),
+                                        )
+                                        .await;
+                                    }
+                                    Err(e) => {
+                                        Self::failed(
+                                            chan,
+                                            format!("Failed to set git settings: {}", e),
+                                        )
+                                        .await;
+                                        return;
+                                    }
+                                }
+                            }
+                            Ok("No") => {
+                                Self::failed(chan, "Git FS Monitor not configured".to_string())
+                                    .await;
+                                return;
+                            }
+                            Ok(_) => unreachable!(),
+                            Err(_) => {
+                                // the sender (terminal) was shut, ignore
+                            }
+                        }
+                    } else {
+                        Self::log_line(&chan, "Git FS Monitor settings set".to_string()).await;
+                    }
+                }
+                Ok(_) => unreachable!(), // the vec of futures has exactly 3 elements
+                Err(e) => {
+                    Self::failed(chan, format!("Failed to get git version: {}", e)).await;
+                    return;
+                }
+            }
+
+            Self::done(chan, "Git FS Monitor enabled".to_string()).await;
+        });
+    }
+}
+
+pub struct DaemonDiagnostic<'a>(pub &'a DaemonPaths);
+
+impl<'a> Diagnostic for DaemonDiagnostic<'a> {
+    const NAME: &'static str = "turbo.daemon";
+
+    fn execute(&self, chan: MpscSender<DiagnosticMessage>) {
+        let paths = self.0.clone();
+        tokio::task::spawn(async move {
+            Self::started(&chan, "Turbo Daemon".to_string()).await;
+            Self::log_line(&chan, "Connecting to daemon...".to_string()).await;
+
+            let pid_path = paths.pid_file.as_std_path().to_owned();
+
+            let connector = DaemonConnector {
+                can_kill_server: false,
+                can_start_server: true,
+                paths,
+            };
+
+            let mut client = match connector.connect().await {
+                Ok(client) => client,
+                Err(e) => {
+                    Self::failed(chan, format!("Failed to connect to daemon: {}", e)).await;
```

Comment:
Should we say something like "failed to connect to daemon, please start it with `turbo daemon start`"? Or even offer to start it for them?